Frends.HTTP.DownloadFile/Frends.HTTP.DownloadFile/Extensions.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Frends.HTTP.DownloadFile.Definitions;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+using System.Text.RegularExpressions;
+
+namespace Frends.HTTP.DownloadFile;
+
+internal static class Extensions
+{
+    internal static void SetHandlerSettingsBasedOnOptions(this HttpClientHandler handler, Options options)
+    {
+        switch (options.Authentication)
+        {
+            case Authentication.WindowsIntegratedSecurity:
+                handler.UseDefaultCredentials = true;
+                break;
+            case Authentication.WindowsAuthentication:
+                var domainAndUserName = options.Username.Split('\\');
+                if (domainAndUserName.Length != 2)
+                    throw new ArgumentException($@"Username needs to be 'domain\username' now it was '{options.Username}'");
+
+                handler.Credentials =
+                    new NetworkCredential(domainAndUserName[1], options.Password, domainAndUserName[0]);
+                break;
+            case Authentication.ClientCertificate:
+                handler.ClientCertificates.AddRange(GetCertificates(options));
+                break;
+        }
+
+        handler.AllowAutoRedirect = options.FollowRedirects;
+        handler.UseCookies = options.AutomaticCookieHandling;
+
+        if (options.AllowInvalidCertificate)
+            handler.ServerCertificateCustomValidationCallback = (a, b, c, d) => true;
+    }
+
+    internal static void SetDefaultRequestHeadersBasedOnOptions(this HttpClient httpClient, Options options)
+    {
+        //Do not automatically set expect 100-continue response header
+        httpClient.DefaultRequestHeaders.ExpectContinue = false;
+        httpClient.DefaultRequestHeaders.TryAddWithoutValidation("content-type", "application/json");
+        httpClient.Timeout = TimeSpan.FromSeconds(Convert.ToDouble(options.ConnectionTimeoutSeconds));
+    }
+
+    // Have to skip these because there's no certificate to use for unit testing.
+    [ExcludeFromCodeCoverage]
```

Comment:
You can create a certificate and test with that one. Example: https://github.com/CommunityHiQ/Frends.Community.Xml/blob/master/Frends.Community.Xml.Tests/XmlTasksTests.cs#L389. You can also just create the certificate locally and push it to the repo for usage in unit tests.